app.js

```diff
@@ -180,9 +180,58 @@ class FilterBox1 {
   }
 }
 
-const filterBox1 = new FilterBox1();
+const filterBox1 = new FilterBoxOpenAndClose();
 
-class FilterBox2 {}
+class FilterByType {
+  constructor() {
+    this.onlineCheckbox = document.querySelector("#onlineChallengeCheckbox");
+    this.onsiteCheckbox = document.querySelector("#onsiteChallengeCheckbox");
+
+    this.onlineCheckbox.addEventListener("change", () => {
+      this.filterChallengesByType();
+    });
+
+    this.onsiteCheckbox.addEventListener("change", () => {
+      this.filterChallengesByType();
+    });
+  }
+
+  filterChallengesByType() {
+    const onlineChecked = this.onlineCheckbox.checked;
+    const onsiteChecked = this.onsiteCheckbox.checked;
+
+    const challengesContainer = document.querySelector("#ourChallenges");
+    const api = new APIadapter();
```

Comment:
If this had been passed into the constructor, instead of creating it here, it would have allowed for dependency injection which opens up for better testing.